IllegalArgumentException when njord.autoPublish=true

Stacktrace:
```
[ERROR] Error closing Njord: Publisher not found: central -> [Help 1]
org.apache.maven.MavenExecutionException: Error closing Njord
    at eu.maveniverse.maven.njord.extension3.NjordSessionLifecycleParticipant.afterSessionEnd (NjordSessionLifecycleParticipant.java:87)
    at org.apache.maven.DefaultMaven.afterSessionEnd (DefaultMaven.java:304)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:272)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: java.lang.IllegalArgumentException: Publisher not found: central
    at eu.maveniverse.maven.njord.shared.impl.DefaultSession.publishSessionArtifactStores (DefaultSession.java:262)
    at eu.maveniverse.maven.njord.extension3.NjordSessionLifecycleParticipant.afterSessionEnd (NjordSessionLifecycleParticipant.java:77)
    at org.apache.maven.DefaultMaven.afterSessionEnd (DefaultMaven.java:304)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:272)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)

```

The project have the following property:

```
  <properties>
    ...
    <njord.autoPublish>true</njord.autoPublish>
    ...
  </properties>
``` 

simply running `mvn validate` cause the issue

--- a/core/src/main/java/eu/maveniverse/maven/njord/shared/impl/DefaultSession.java
+++ b/core/src/main/java/eu/maveniverse/maven/njord/shared/impl/DefaultSession.java
@@ -32,6 +32,7 @@ import java.util.Map;
 import java.util.Optional;
 import java.util.UUID;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.Supplier;
@@ -156,13 +157,11 @@ public class DefaultSession extends CloseableConfigSupport<SessionConfig> implem
         }
     }
 
-    @SuppressWarnings("unchecked")
     @Override
     public ArtifactStore getOrCreateSessionArtifactStore(String uri) {
         requireNonNull(uri);
 
-        ConcurrentHashMap<String, String> sessionBoundStore = (ConcurrentHashMap<String, String>)
-                config.session().getData().computeIfAbsent(sessionBoundStoreKey, ConcurrentHashMap::new);
+        ConcurrentMap<String, String> sessionBoundStore = getSessionBoundStore();
         String storeName = sessionBoundStore.computeIfAbsent(uri, k -> {
             try {
                 String artifactStoreName;
@@ -228,15 +227,16 @@ public class DefaultSession extends CloseableConfigSupport<SessionConfig> implem
         }
     }
 
-    @SuppressWarnings("unchecked")
     @Override
     public int publishSessionArtifactStores() {
         int published = 0;
         for (Supplier<Integer> callable : derivedPublishSessionArtifactStores) {
             published += callable.get();
         }
-        ConcurrentHashMap<String, String> sessionBoundStore = (ConcurrentHashMap<String, String>)
-                config.session().getData().computeIfAbsent(sessionBoundStoreKey, ConcurrentHashMap::new);
+        ConcurrentMap<String, String> sessionBoundStore = getSessionBoundStore();
+        if (sessionBoundStore.isEmpty()) {
+            return published;
+        }
         AtomicInteger result = new AtomicInteger(published);
         Optional<String> pno = artifactDeployerRedirector.getArtifactStorePublisherName(config);
         if (pno.isPresent()) {
@@ -262,20 +262,18 @@ public class DefaultSession extends CloseableConfigSupport<SessionConfig> implem
                 throw new IllegalArgumentException("Publisher not found: " + publisherName);
             }
         } else {
-            throw new IllegalStateException("Njord autoPublish enabled, but publisher not set");
+            throw new IllegalStateException("Publisher name was not specified nor could be discovered");
         }
         return result.get();
     }
 
-    @SuppressWarnings("unchecked")
     @Override
     public int dropSessionArtifactStores() {
         int dropped = 0;
         for (Supplier<Integer> callable : derivedDropSessionArtifactStores) {
             dropped += callable.get();
         }
-        ConcurrentHashMap<String, String> sessionBoundStore = (ConcurrentHashMap<String, String>)
-                config.session().getData().computeIfAbsent(sessionBoundStoreKey, ConcurrentHashMap::new);
+        ConcurrentMap<String, String> sessionBoundStore = getSessionBoundStore();
         AtomicInteger result = new AtomicInteger(dropped);
         sessionBoundStore.values().forEach(n -> {
             try {
@@ -292,5 +290,14 @@ public class DefaultSession extends CloseableConfigSupport<SessionConfig> implem
     @Override
     protected void doClose() throws IOException {
         internalArtifactStoreManager.close();
+    }
+
+    /**
+     * Returns map of "Njord URI" to "storeName" that were created in current session.
+     */
+    @SuppressWarnings("unchecked")
+    private ConcurrentMap<String, String> getSessionBoundStore() {
+        return (ConcurrentHashMap<String, String>)
+                config.session().getData().computeIfAbsent(sessionBoundStoreKey, ConcurrentHashMap::new);
     }
 }

--- a/extension/src/main/java/eu/maveniverse/maven/njord/extension3/NjordSessionLifecycleParticipant.java
+++ b/extension/src/main/java/eu/maveniverse/maven/njord/extension3/NjordSessionLifecycleParticipant.java
@@ -76,7 +76,9 @@ public class NjordSessionLifecycleParticipant extends AbstractMavenLifecyclePart
                     } else if (njordSession.config().autoPublish()) {
                         int published = njordSession.publishSessionArtifactStores();
                         if (published != 0) {
-                            logger.info("Published {} stores created in this session", published);
+                            logger.info("Auto publish: Published {} stores created in this session", published);
+                        } else {
+                            logger.info("Auto publish: No stores created in this session");
                         }
                     }
                     logger.info("Njord session closed");

--- a/plugin/src/main/java/eu/maveniverse/maven/njord/plugin3/PublisherSupportMojo.java
+++ b/plugin/src/main/java/eu/maveniverse/maven/njord/plugin3/PublisherSupportMojo.java
@@ -120,7 +120,7 @@ public abstract class PublisherSupportMojo extends NjordMojoSupport {
     protected ArtifactStorePublisher getArtifactStorePublisher(Session ns) throws MojoFailureException {
         Optional<String> publisherName = getArtifactStorePublisherName(ns);
         if (publisherName.isEmpty()) {
-            throw new MojoFailureException("Publisher name was not specified nor could be found");
+            throw new MojoFailureException("Publisher name was not specified nor could be discovered");
         }
         String publisher = publisherName.orElseThrow();
         Optional<ArtifactStorePublisher> po = ns.selectArtifactStorePublisher(publisher);

